FIX remove action of normalize_components in SparsePCA

glemaitre opened the pull request:
I forgot to remove the following line when dealing with the deprecation.
Note that `normalize_components` should not have any effect and the algorithm behave like if `normalize_components=True`

glemaitre committed:
FIX remove action of normalize_components in SparsePCA

glemaitre commented:
@rth this is an easy one

rth reviewed (approved):
Looks good. Maybe also add SparcePCA to "Changed models" in what's new because as of 0.22 `normalize_components` is deprecated, still exists and has no effect anymore.

glemaitre committed:
address roman comments

thomasjpfan reviewed (approved):
The history of this parameter is quite something. We added it in `0.20` (for backwards compatibility) just so we can deprecate it in `0.22`.

glemaitre commented:
> The history of this parameter is quite something. We added it in 0.20 (for backwards compatibility) just so we can deprecate it in 0.22.

Actually we added it to keep a backwards compatibility that was certainly a bug :)

thomasjpfan commented:
Thank you! @glemaitre
